src/main/java/ladder/domain/Ladder.java

```diff
@@ -0,0 +1,31 @@
+package ladder.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import ladder.domain.generator.LineGenerator;
+
+public class Ladder {
+
+    private final List<Line> lines;
+
+    public Ladder(final LineGenerator lineGenerator, final Players players, final Height height) {
+        this.lines = generateLines(lineGenerator, players, height);
```

Comment:
Ladder를 생성하기 위해선 매번  final LineGenerator lineGenerator, final Players players, final Height height를 알아야겠군요.

생성자에 로직을 빼고, 인자로 들어오는 값만 넣어주는건 어떨까요?
생성자는 특수한 메서드인 만큼, 완전한 인자들을 제공받아야합니다. 

물론 아래와 같은 코드는 제외하고요. 
```suggestion
public class Ladder {

  public Ladder(final LineGenerator lineGenerator, final Players players, final Height height) {
    this(generateLines(lineGenerator, players, height));
  }

  private Ladder(List<Line> lines) {
    this.lines = lines;
  }
```

Reply:
생성자에서 진행되는 `generateLines` 역할을 `LadderController`로 넘겨줄 수 있을 것 같아요! 생성자에 대해서도 다시 한번 고민해볼 수 있었습니다. 좋은 리뷰 감사합니다 :)